src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/AuthorizationBehavior.cs

```diff
@@ -17,12 +19,21 @@ internal sealed class AuthorizationBehavior
         private static readonly ServiceAuthorizationManager s_defaultServiceAuthorizationManager = new ServiceAuthorizationManager();
         private ReadOnlyCollection<IAuthorizationPolicy> _externalAuthorizationPolicies;
         private ServiceAuthorizationManager _serviceAuthorizationManager;
+        private IAuthorizationService _authorizationService;
 
         private AuthorizationBehavior() { }
 
         public async ValueTask<MessageRpc> AuthorizeAsync(MessageRpc rpc)
         {
             // TODO: Events
+            ClaimsPrincipal claimsPrincipal = rpc.Request.Properties.Security?.ServiceSecurityContext?.ClaimsPrincipal;
+            AuthorizationPolicy authorizationPolicy = rpc.Operation.AuthorizationPolicy;
```

Comment:
We should move the call to `DispatchOperationRuntime.SetClaimsPrincipalToOperationContext` to happen either as part of this method, or before this method is called. Then you can get the ClaimsPrincipal from rpc.OperationContext.ClaimsPrincipal

Reply:
done in https://github.com/CoreWCF/CoreWCF/pull/850/commits/0da8aca434f3bcb07d3fd55955a69d39cb3cd352